pkgs/development/libraries/openssl/1.0.2.x.nix

```diff
@@ -18,7 +18,9 @@ stdenv.mkDerivation rec {
     sha1 = "d01d17b44663e8ffa6a33a5a30053779d9593c3d";
   };
 
-  patches = optional stdenv.isCygwin ./1.0.1-cygwin64.patch;
+  patches =
+    [ ./openssl-1.0.0d-windres.patch ]
```

Comment:
That's a good point.

Upstream doesn't carry different patch sets for different platforms either, but you're correct that an optional wouldn't hurt at all. However, I'm unsure which exact condition to test against to make sure it is caught in all cases where required. So it seems simpler just to apply it uniformly, as it won't hurt any of the other platforms where windres is not used.